rbd/clone_image_by_id.go

```diff
@@ -0,0 +1,46 @@
+//go:build !(nautilus || octopus || pacific || quincy || reef) && ceph_preview
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <errno.h>
+// #include <stdlib.h>
+// #include <rados/librados.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/ceph/go-ceph/rados"
+)
+
+// CloneImageByID creates a clone of the image from a snapshot with the given
+// ID in the provided io-context with the given name and image options.
+//
+// Implements:
+//
+//	int rbd_clone4(rados_ioctx_t p_ioctx, const char *p_name,
+//	               uint64_t p_snap_id, rados_ioctx_t c_ioctx,
+//	               const char *c_name, rbd_image_options_t c_opts);
+func CloneImageByID(ioctx *rados.IOContext, parentName string, snapID uint64,
```

Comment:
Should we call it `CloneImageBySnapID`?

Reply:
There is `CloneImage` which also uses a snapshot (by name). If you really prefer `CloneImageBySnapID()`, I don't mind to change it.

Reply:
FWIW either name is fine with me. Ping me for re-review one you all have decided what to use.

Reply:
> There is `CloneImage` which also uses a snapshot (by name).

Ah, we already have something similar. I didn't bother to look but that's what I had in mind while suggesting the name change.

> If you really prefer `CloneImageBySnapID()`, I don't mind to change it.

Let's not change it then.